addons/binding/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/handler/TiVoHandler.java

```diff
@@ -0,0 +1,794 @@
+        }
+
+        value = conf.get(CONFIG_CMD_WAIT_INTERVAL);
+        if (value != null) {
+            tivoConfig.setCfgCmdWait(confValueToInt(value));
+        }
+
+        value = conf.get(CONFIG_CH_START);
+        if (value != null) {
+            tivoConfig.setCfgMinChannel(confValueToInt(value));
+        }
+
+        value = conf.get(CONFIG_CH_END);
+        if (value != null) {
+            tivoConfig.setCfgMaxChannel(confValueToInt(value));
+        }
+
+        value = conf.get(CONFIG_IGNORE_SCAN);
+        if (value != null) {
+            tivoConfig.setCfgIgnoreChannelScan(confValueToBoolean(value));
+        }
+
+        value = getThing().getUID();
+        if (value != null) {
+            tivoConfig.setCfgIdentifier(String.valueOf(value));
+        }
+
+        value = conf.get(CONFIG_IGNORE_CHANNELS);
+        if (value != null) {
+            tivoConfig.setCfgIgnoreChannels(chParseIgnored(String.valueOf(conf.get(CONFIG_IGNORE_CHANNELS)),
+                    tivoConfig.getCfgMinChannel(), tivoConfig.getCfgMaxChannel()));
+        }
+
+        logger.debug("TivoConfigData Obj: '{}'", tivoConfig.toString());
+        tivoCfgData = tivoConfig;
+
+        if (myTivoService == null) {
+            myTivoService = new TivoStatusProvider(tivoCfgData, this, false);
+        }
+
+        if (tivoConfig.getCfgIgnoreChannelScan()) {
+            startChannelScan();
+
+        } else if (tivoConfig.isCfgPollChanges()) {
+            startPollStatus();
+        }
+
+        updateStatus(ThingStatus.OFFLINE);
+        myTivoService.statusRefresh();
+        logger.debug("Initializing a TiVo handler for thing '{}' - finished!", getThing().getUID());
+
+    }
+
+    @Override
+    public void dispose() {
+
+        logger.debug("Disposing of a TiVo handler for thing '{}'", getThing().getUID());
+
+        if (refreshJob != null) {
+            refreshJob.cancel(false);
+        }
+        if (chScanJob != null) {
+            logger.warn("'{}' - Channel Scan cancelled by dispose()", getThing().getUID());
+            chScanJob.cancel(false);
+        }
+
+        while (chScanJob != null && !chScanJob.isDone()) {
+            try {
+                TimeUnit.MILLISECONDS.sleep(tivoCfgData.getCfgCmdWait());
+            } catch (InterruptedException e) {
+                logger.debug("Disposing '{}' while waiting for 'channelScanJob' to end error: '{}' ",
+                        getThing().getUID(), e.getMessage());
+            }
+        }
+
+        // Ensure we close any open socket connections
+        if (myTivoService != null) {
+            myTivoService.connTivoConnectRetry(false);
+            myTivoService = null;
+        }
+
+    }
+
+    /**
+     * {@link startPollStatus} scheduled job to poll for changes in state.
+     */
+    private void startPollStatus() {
+
+        int firstStartDelay = tivoCfgData.getCfgPollInterval();
+
+        refreshJob = scheduler.scheduleAtFixedRate(new Runnable() {
```

Comment:
Please use [scheduleWithFixedDelay](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleWithFixedDelay(java.lang.Runnable,%20long,%20long,%20java.util.concurrent.TimeUnit)) which is preferred over scheduleAtFixedRate, see [Coding Guidelines D.3](http://docs.openhab.org/developers/development/guidelines.html#d-runtime-behavior)